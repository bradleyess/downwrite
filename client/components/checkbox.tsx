import * as React from "react";
import classNames from "../utils/classnames";

export default function CheckboxInput({
  className,
  ...props
}: React.InputHTMLAttributes<HTMLInputElement>): JSX.Element {
  const cx = classNames("UICheck", className);

  return React.createElement("input", { type: "checkbox", ...props, className: cx });
}
