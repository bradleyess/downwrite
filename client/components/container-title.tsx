import styled from "styled-components";
import { fonts } from "../utils/defaultStyles";

const ContainerTitle = styled.h1`
  font-weight: 300;
  font-size: 18px;
  font-family: ${fonts.sans};

  @media (min-width: 57.75rem) {
    font-size: 24px;
  }
`;

export default ContainerTitle;
