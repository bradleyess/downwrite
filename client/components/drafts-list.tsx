import * as React from "react";
import { ILocalDraft } from "../hooks/local-draft";

interface IDraftListProps {
  drafts: ILocalDraft[];
  onRemove(d: ILocalDraft): void;
}

export default function DraftList(props: IDraftListProps): JSX.Element {
  return props.drafts.length > 0 ? (
    <ul>
      {props.drafts.map((draft, i) => {
        return (
          <li key={draft.id || i}>
            <span>{draft.title} </span>
            <button onClick={() => console.log(draft)}>Edit</button>
            <button onClick={() => props.onRemove(draft)}>Remove</button>
          </li>
        );
      })}
    </ul>
  ) : null;
}
