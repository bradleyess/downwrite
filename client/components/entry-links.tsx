import * as React from "react";
import Link from "next/link";
import { UrlObject } from "url";

interface ICardLinks {
  id: string;
  title?: string;
  style?: React.CSSProperties;
  className?: string;
}

export function EditLink(props: ICardLinks): JSX.Element {
  const url: UrlObject = { pathname: "/edit", query: { id: props.id } };

  return (
    <Link prefetch passHref href={url}>
      <a className={props.className} style={props.style}>
        {props.title || "Edit"}
      </a>
    </Link>
  );
}

export function PreviewLink(props: ICardLinks): JSX.Element {
  const url: UrlObject = { pathname: "/preview", query: { id: props.id } };

  return (
    <Link prefetch passHref href={url}>
      <a className={props.className} style={props.style}>
        Preview
      </a>
    </Link>
  );
}
