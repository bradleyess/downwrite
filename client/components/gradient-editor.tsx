import * as React from "react";
import Avatar from "./avatar";

interface IGradientEditorProps {
  initialColors: any;
}

interface IGradientEditorState {
  colors: string[];
}

export default class GradientEditor extends React.Component<
  IGradientEditorProps,
  IGradientEditorState
> {
  state = {
    colors: this.props.initialColors
  };

  render() {
    const { colors } = this.state;
    return <Avatar colors={colors} />;
  }
}
