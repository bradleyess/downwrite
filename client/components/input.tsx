import * as React from 'react';
import styled from 'styled-components';

const StyledInput = styled.input`
  display: block;
  width: 100%;
  appearance: none;
  font-weight: 700;
  color: inherit;
  font-family: inherit;
  background: none;
  font-size: 150%;
  border-width: 0px;
  border-bottom: 1px;
  border-style: solid;
  border-radius: 0px;
  border-color: ${props => props.theme.inputBorder};
  padding-left: 0px;
  padding-right: 0px;
  padding-top: 8px;
  padding-bottom: 8px;
  outline: none;
  transition: border-color 250ms linear;

  &:focus {
    border-color: ${props => props.theme.link};
  }
`;

interface IInputProps {
  type: string;
  inputRef?: React.RefObject<any>;
  onChange: React.ChangeEventHandler<HTMLInputElement>;
}

export default class Input extends React.Component<IInputProps, any> {
  static defaultProps = {
    type: 'text'
  };

  render() {
    const { onChange, inputRef, ...args } = this.props;

    return <StyledInput ref={inputRef} onChange={e => onChange(e)} {...args} />;
  }
}
