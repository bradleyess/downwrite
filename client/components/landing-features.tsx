import * as React from 'react';
import styled from 'styled-components';

const FeatureContainer = styled.section`
  max-width: 33rem;
  margin: 1rem auto 0;
  font-size: 14px;
  text-align: left;

  h2 {
    font-size: 18px;
    margin-bottom: 4px;
  }

  p:not(:last-of-type) {
    margin-bottom: 16px;
  }
`;

export default () => (
  <FeatureContainer>
    <h2>Focus on Markdown</h2>
    <p>
      Writing should be easy. But as each tool, each static site builder comes and
      falls out of popularity or gets shut down, <b>**markdown**</b> remains the
      central and portbale format.
    </p>

    <p>
      The goal of building Downwrite was to create a place to write and share content
      with that universal format; to be able to import and export in markdown, to
      write in markdown and share your work.
    </p>

    <p>
      <i>_Sign up below and get to writing._</i>
    </p>
  </FeatureContainer>
);
