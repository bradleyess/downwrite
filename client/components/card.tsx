import * as React from "react";
import * as Draft from "draft-js";
import distance from "date-fns/distance_in_words_to_now";
import { IPost } from "downwrite";
import { EditLink, PreviewLink } from "./entry-links";

export interface ICardProps {
  title: string;
  content: Draft.RawDraftContentState;
  excerpt?: string;
  id: string;
  dateAdded: Date;
  onDelete?: ({ id }: Partial<IPost>) => void;
  public: boolean;
}

export default function Card(props: ICardProps) {
  function onDelete() {
    props.onDelete({ id: props.id, title: props.title });
  }

  return (
    <div className="Sheet Card" data-testid="CARD">
      <header className="CardHeader">
        <h2 className="CardTitle" data-testid="CARD_TITLE">
          <EditLink title={props.title} id={props.id} />
        </h2>
        <small className="CardMeta">added {distance(props.dateAdded)} ago</small>
      </header>
      <footer className="CardTray">
        <div className="links" data-testid="CARD_EXCERPT">
          <EditLink style={{ marginRight: 8 }} id={props.id} />
          {props.public && <PreviewLink style={{ marginRight: 8 }} id={props.id} />}
        </div>
        {props.onDelete && (
          <button
            className="CardDeleteButton"
            data-testid="CARD_DELETE_BUTTON"
            onClick={onDelete}>
            Delete
          </button>
        )}
      </footer>
    </div>
  );
}
