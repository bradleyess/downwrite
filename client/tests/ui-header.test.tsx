import * as React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { UIHeader } from "../components/header";
import { LinkProps } from "next/link";
import { MockAuthProvider } from "../components/auth";

jest.mock("universal-cookie", () => {
  return class Cookie {};
});
jest.mock("jwt-decode");

jest.mock("next/link", () => {
  return jest.fn((props: React.PropsWithChildren<LinkProps>) => (
    <>{props.children}</>
  ));
});

let { getByTestId, container } = render(
  <MockAuthProvider>
    <UIHeader />
  </MockAuthProvider>
);

// NOTE: test broken by upgrading @testing-library
describe("Header Component", () => {
  it("contains application name", () => {
    expect(getByTestId("APP_HEADER_TITLE")).toHaveTextContent("Downwrite");
    expect(getByTestId("APP_HEADER")).toBeTruthy();
  });

  it("matches snapshot", () => {
    expect(container.firstChild).toMatchSnapshot();
  });
});
