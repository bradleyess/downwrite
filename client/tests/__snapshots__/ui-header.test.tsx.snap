// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component matches snapshot 1`] = `null`;
